src/Provider/IP2Location/Tests/IP2LocationTest.php

```diff
@@ -0,0 +1,131 @@
+    {
+        if (!isset($_SERVER['IP2LOCATION_API_KEY'])) {
+            $this->markTestSkipped('You need to configure the IP2LOCATION_API_KEY value in phpunit.xml');
+        }
+
+        $provider = new IP2Location($this->getHttpClient($_SERVER['IP2LOCATION_API_KEY']), $_SERVER['IP2LOCATION_API_KEY']);
+        $results = $provider->geocodeQuery(GeocodeQuery::create('74.125.45.100'));
+
+        $this->assertInstanceOf('Geocoder\Model\AddressCollection', $results);
+        $this->assertCount(1, $results);
+
+        /** @var Location $result */
+        $result = $results->first();
+        $this->assertInstanceOf('\Geocoder\Model\Address', $result);
+        $this->assertEquals(36.154, $result->getCoordinates()->getLatitude(), '', 0.001);
+        $this->assertEquals(-95.9928, $result->getCoordinates()->getLongitude(), '', 0.001);
+        $this->assertEquals(74101, $result->getPostalCode());
+        $this->assertEquals('Tulsa', $result->getLocality());
+        $this->assertCount(1, $result->getAdminLevels());
+        $this->assertEquals('Oklahoma', $result->getAdminLevels()->get(1)->getName());
+        $this->assertEquals('United States', $result->getCountry()->getName());
+        $this->assertEquals('US', $result->getCountry()->getCode());
+    }
+
+    public function testGeocodeWithRealIPv6()
+    {
+        if (!isset($_SERVER['IP2LOCATION_API_KEY'])) {
+            $this->markTestSkipped('You need to configure the IP2LOCATION_API_KEY value in phpunit.xml');
+        }
+
+        $provider = new IP2Location($this->getHttpClient($_SERVER['IP2LOCATION_API_KEY']), $_SERVER['IP2LOCATION_API_KEY']);
+        $results = $provider->geocodeQuery(GeocodeQuery::create('::ffff:74.125.45.100'));
+
+        $this->assertInstanceOf('Geocoder\Model\AddressCollection', $results);
+        $this->assertCount(1, $results);
+
+        /** @var Location $result */
+        $result = $results->first();
+        $this->assertInstanceOf('\Geocoder\Model\Address', $result);
+        $this->assertEquals(36.154, $result->getCoordinates()->getLatitude(), '', 0.001);
+        $this->assertEquals(-95.9928, $result->getCoordinates()->getLongitude(), '', 0.001);
+        $this->assertEquals(74101, $result->getPostalCode());
+        $this->assertEquals('Tulsa', $result->getLocality());
+        $this->assertCount(1, $result->getAdminLevels());
+        $this->assertEquals('Oklahoma', $result->getAdminLevels()->get(1)->getName());
+        $this->assertEquals('United States', $result->getCountry()->getName());
+        $this->assertEquals('US', $result->getCountry()->getCode());
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\UnsupportedOperation
+     * @expectedExceptionMessage The IP2Location provider is not able to do reverse geocoding.
+     */
+    public function testReverse()
+    {
+        $provider = new IP2Location($this->getMockedHttpClient(), 'api_key');
+        $provider->reverseQuery(ReverseQuery::fromCoordinates(0, 0));
+    }
```

Comment:
Can we have some tests for errors?
- invalid api key
- invalid ip format

etc?